flake.nix

```diff
@@ -9,79 +9,81 @@
     };
 
-    naersk = {
-      url = "https://flakehub.com/f/nix-community/naersk/0.1.*";
-      inputs.nixpkgs.follows = "nixpkgs";
-    };
+    crane.url = "https://flakehub.com/f/ipetkov/crane/0";
   };
 
-  outputs = { self, ... }@inputs:
+  outputs = inputs:
     let
       forSystems = s: f: inputs.nixpkgs.lib.genAttrs s (system: f rec {
         inherit system;
-        pkgs = import inputs.nixpkgs { inherit system; overlays = [ self.overlays.default ]; };
+        pkgs = import inputs.nixpkgs {
+          inherit system;
+          overlays = [ inputs.self.overlays.default ];
+        };
       });
 
       forAllSystems = forSystems [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
-
-      fenixToolchain = system: with inputs.fenix.packages.${system};
-        combine ([
-          stable.clippy
-          stable.rustc
-          stable.cargo
-          stable.rustfmt
-          stable.rust-src
-        ] ++ inputs.nixpkgs.lib.optionals (system == "x86_64-linux") [
-          targets.x86_64-unknown-linux-musl.stable.rust-std
-        ] ++ inputs.nixpkgs.lib.optionals (system == "aarch64-linux") [
-          targets.aarch64-unknown-linux-musl.stable.rust-std
-        ]);
-
     in
     {
-      overlays.default = final: prev: rec {
-        fh =
-          let
-            rustToolchain = fenixToolchain final.stdenv.hostPlatform.system;
-            naerskLib = final.callPackage inputs.naersk {
-              cargo = rustToolchain;
-              rustc = rustToolchain;
+      overlays.default = final: prev:
+        let
+          system = prev.stdenv.hostPlatform.system;
+          staticTarget =
+            {
+              "aarch64-linux" = "aarch64-unknown-linux-musl";
+              "x86_64-linux" = "x86_64-unknown-linux-musl";
+            }.${system} or null;
+        in
+        rec {
+          fh =
+            let
+              craneLib = inputs.crane.mkLib prev;
+            in
+            craneLib.buildPackage {
+              name = "fh";
+              src = inputs.self;
+
+              doCheck = true;
+
+              nativeBuildInputs = with final; [
+                pkg-config
+                rustPlatform.bindgenHook
+                installShellFiles
+              ];
+
+              buildInputs = with final; [
+                gcc.cc.lib
+              ]
+              ++ lib.optionals (stdenv.isDarwin) (with darwin.apple_sdk.frameworks; [
+                Security
+                SystemConfiguration
+              ]);
+
+              postInstall = ''
+                installShellCompletion --cmd fh \
+                  --bash <("$out/bin/fh" completion bash) \
+                  --zsh <("$out/bin/fh" completion zsh) \
+                  --fish <("$out/bin/fh" completion fish)
+              '';
+
+              env = {
+                SSL_CERT_FILE = "${final.cacert}/etc/ssl/certs/ca-bundle.crt";
+                LIBCLANG_PATH = "${final.libclang.lib}/lib";
```

Comment:
This _should_ be handled by the `bindgenHook`. Can you remove it and verify?

Reply:
Yep, seems fine

Reply:
Playing with this locally, it looks like `LIBCLANG_PATH` is safe to remove. I don't have a Mac, but I suspect that `NIX_CFLAGS_COMPILE` is also safe to remove. `SSL_CERT_FILE`, however, is load-bearing.